services/QuillLMS/app/controllers/auth/classlink_controller.rb

```diff
@@ -46,5 +49,12 @@ def classlink
     private def user
       auth_credential&.user
     end
+
+    private def verify_teacher_if_necessary
+      return if user.teacher_verification_status == TeacherInfo::APPROVED || user.admin_approval_status == AdminInfo::APPROVED
```

Comment:
I think we want to keep teacher auto-approval and admin auto-approval in a single code location. Here's the admin auto-approval code: https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/services/classlink_integration/instructor_linker.rb#L32

Note that the admin approval code happens at Rostering time. It may make sense to move admin- and teacher- auto approvals to sign-in time, given the product ask.

Reply:
Ah, I didn't know that existed. I agree they should be in the same place. I'm going to check in with Nattalie and confirm that it's okay if this happens at rostering instead of sign-in.